test/Profiler/coverage_errors.swift

```diff
@@ -0,0 +1,411 @@
+    try test5()          // CHECK-NEXT: [[@LINE]]:16 -> [[@LINE+6]]:4 : (0 - 1)
+    do {                 // CHECK-NEXT: [[@LINE]]:8  -> [[@LINE+2]]:6 : (0 - 1)
+      throw SomeErr.Err1
+    } catch {            // CHECK-NEXT: [[@LINE]]:13 -> [[@LINE+2]]:6 : 2
+      return 0
+    }
+  } catch {              // CHECK-NEXT: [[@LINE]]:11 -> [[@LINE+2]]:4 : 3
+    return 1
+  }
+}                        // CHECK-NEXT: }
+
+// Test coverage with a do-catch inside of a repeat-while
+// CHECK-LABEL: sil_coverage_map {{.*}} "$s15coverage_errors6test10SiyF"
+func test10() -> Int {
+  repeat {               // CHECK: [[@LINE]]:10 {{.*}} : 1
+    do {
+      throw SomeErr.Err1
+    } catch {            // CHECK: [[@LINE]]:13 {{.*}} : 2
+      return 0
+    }                    // CHECK-NOT: [[@LINE]]:6 ->
+
+  } while false          // CHECK: [[@LINE]]:11 {{.*}} : (1 - 2)
+  return 1
+}
+
+// Test coverage with a break inside a do-catch inside of a repeat-while
+// CHECK-LABEL: sil_coverage_map {{.*}} "$s15coverage_errors6test11SiyF"
+func test11() -> Int { // CHECK-NEXT: [[@LINE]]:22 -> [[@LINE+11]]:2 : 0
+  repeat {             // CHECK-NEXT: [[@LINE]]:10 -> [[@LINE+8]]:4 : 1
+    do {               // CHECK-NEXT: [[@LINE]]:8  -> [[@LINE+2]]:6 : 1
+      try test5()      // CHECK-NEXT: [[@LINE]]:18 -> [[@LINE+1]]:6 : (1 - 2)
+    } catch {          // CHECK-NEXT: [[@LINE]]:13 -> [[@LINE+2]]:6 : 3
+      break
+    }                  // CHECK-NEXT: [[@LINE]]:6   -> [[@LINE+3]]:4  : (1 - 2)
+                       // FIXME: This exit counter is wrong (rdar://118472537)
+                       // CHECK-NEXT: [[@LINE+1]]:4 -> [[@LINE+2]]:11 : (0 - 2)
+  } while false        // CHECK-NEXT: [[@LINE]]:11  -> [[@LINE]]:16   : (1 - 3)
+  return 1
+}                      // CHECK-NEXT: }
+
+// rdar://41010883 – Make sure we don't introduce an empty unreachable region.
+// CHECK-LABEL: sil_coverage_map {{.*}} "$s15coverage_errors6test12SiyKF"
+func test12() throws -> Int { // CHECK-NEXT: [[@LINE]]:29 -> [[@LINE+7]]:2  : 0
+  do {                        // CHECK-NEXT: [[@LINE]]:6  -> [[@LINE+3]]:4  : 0
+    try test5()               // CHECK-NEXT: [[@LINE]]:16 -> [[@LINE+1]]:13 : (0 - 1)
+    return 1
+  } catch is SomeErr {        // CHECK-NEXT: [[@LINE]]:22 -> [[@LINE+2]]:4 : 2
+    throw SomeErr.Err1
+  }
+}                             // CHECK-NEXT: }
+
+// CHECK-LABEL: sil_coverage_map {{.*}} "$s15coverage_errors6test13SiyF"
+func test13() -> Int {      // CHECK-NEXT: [[@LINE]]:22 -> [[@LINE+6]]:2 : 0
+  do {                      // CHECK-NEXT: [[@LINE]]:6  -> [[@LINE+2]]:4 : 0
+    return try throwingFn() // Note we don't emit a region here because it would be empty.
+  } catch {                 // CHECK-NEXT: [[@LINE]]:11 -> [[@LINE+2]]:4 : 2
+    return 0
+  }
+}                           // CHECK-NEXT: }
+
+func takesInts(_ x: Int, _ y: Int) {}
+
+// The throwing expr is nested here, the region starts after the throwing expr.
```

Comment:
Another to consider here would be eg.
```
func foo() throws -> Int { return  1}
func foo2(_ arg: Int, _ arg2: Int, _ arg3: Int) throws -> Int { return 2 }
func bar() throws -> Int {
  // code here
  try foo2(
    1,
    foo(),
    2
  )
  // more here
}
```

The call to `foo2` isn't executed if `foo()` throws. Whether that's worth differentiating... 🤷

EDIT: Ah, you have this case further down in `test20`, could be worth a TODO there.

Reply:
Yeah I'll look into this as a follow up